core/cat/memory/vector_memory.py

```diff
@@ -129,7 +132,28 @@ def create_collection_if_not_exists(self):
                 log(f'Collection "{self.collection_name}" has the same embedder', "INFO")
             else:
                 log(f'Collection "{self.collection_name}" has different embedder', "WARNING")
-                # TODO: dump collection on disk before deleting, so it can be recovered
+                # dump collection on disk before deleting, so it can be recovered
+                # TODO: it needs a code refactoring
+
+                #dormouse = "dormouse/"
+                if os.path.isdir("dormouse/"):
```

Comment:
I propose to move the snapshot procedure in a separate method in the class

Reply:
Done in core/cat/memory/vector_memory.py from row 253